pkg/controllermanager/controller/plant/plant_control.go

```diff
@@ -0,0 +1,347 @@
+	)
+
+	if plant.DeletionTimestamp != nil {
+		return c.delete(plant, logger)
+	}
+
+	return c.reconcile(ctx, plant, key, logger)
+}
+
+func (c *defaultPlantControl) conditionThresholdsToProgressingMapping() map[gardencorev1alpha1.ConditionType]time.Duration {
+	out := make(map[gardencorev1alpha1.ConditionType]time.Duration)
+	for _, threshold := range c.config.Controllers.ShootCare.ConditionThresholds {
+		out[gardencorev1alpha1.ConditionType(threshold.Type)] = threshold.Duration.Duration
+	}
+	return out
+}
+
+func (c *defaultPlantControl) updatePlantConditions(plant *gardencorev1alpha1.Plant, conditions ...gardencorev1alpha1.Condition) (*gardencorev1alpha1.Plant, error) {
+	newPlant, err := kutil.TryUpdatePlantConditions(c.k8sGardenClient.GardenCore(), retry.DefaultBackoff, plant.ObjectMeta,
+		func(plant *gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error) {
+			plant.Status.Conditions = conditions
+			return plant, nil
+		})
+
+	return newPlant, err
+}
+
+func (c *defaultPlantControl) reconcile(ctx context.Context, plant *gardencorev1alpha1.Plant, key string, logger logrus.FieldLogger) error {
+	_, err := kutil.TryUpdatePlantStatusWithEqualFunc(c.k8sGardenClient.GardenCore(), retry.DefaultBackoff, plant.ObjectMeta,
+		func(p *gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error) {
+			if finalizers := sets.NewString(p.Finalizers...); !finalizers.Has(FinalizerName) {
+				finalizers.Insert(FinalizerName)
+				p.Finalizers = finalizers.UnsortedList()
+			}
+			return p, nil
+		}, func(cur, updated *gardencorev1alpha1.Plant) bool {
+			return sets.NewString(cur.Finalizers...).Has(FinalizerName)
+		})
+	if err != nil {
+		return err
+	}
+
+	var (
+		newConditions               = helper.MergeConditions(plant.Status.Conditions, helper.InitCondition(gardencorev1alpha1.PlantAPIServerAvailable), helper.InitCondition(gardencorev1alpha1.PlantEveryNodeReady))
+		conditionAPIServerAvailable = newConditions[0]
+		conditionEveryNodeReady     = newConditions[1]
+	)
+
+	defer func() {
+		if _, err := c.updateConditions(plant, conditionAPIServerAvailable, conditionEveryNodeReady); err != nil {
+			logger.Errorf("Failed to update the conditions : %+v", err)
+		}
+	}()
+
+	kubeconfigSecret, err := c.secretsLister.Secrets(plant.Spec.SecretRef.Namespace).Get(plant.Spec.SecretRef.Name)
+	if err != nil {
+		return err
+	}
+	kubeconfig, ok := kubeconfigSecret.Data["kubeconfig"]
+	if !ok {
+		message := "Plant secret needs to contain a valid kubeconfig."
+		conditionAPIServerAvailable = helper.UpdatedCondition(conditionAPIServerAvailable, corev1.ConditionFalse, "APIServerDown", message)
+		conditionEveryNodeReady = helper.UpdatedCondition(conditionEveryNodeReady, corev1.ConditionFalse, "Nodes not reachable", message)
+		resetClients(c, key)
+
+		return c.updateStatus(plant, &StatusCloudInfo{}, conditionAPIServerAvailable, conditionEveryNodeReady)
+	}
+
+	secretChecksum := utils.ComputeSHA256Hex([]byte(strings.TrimSpace(string(kubeconfig))))
```

Comment:
As disussed in person, let's not use this checksum and let's not store the clients for all plants on the controller. Instead, let's just initialize it here in every reconciliation loop (like in all other controllers). Additionally, let's establish a `WATCH` on `Secret`s that requeue those `Plant`s that are referring to them.